src/high-level/src/KinDynComputations.cpp

```diff
@@ -905,6 +943,127 @@ bool KinDynComputations::getFrameFreeFloatingJacobian(const FrameIndex frameInde
+
+    return getRelativeJacobianExplicit(refFrameIndex, frameIndex, expressedOriginFrame, expressedOrientationFrame, outJacobian);
+}
+
+bool KinDynComputations::getRelativeJacobianExplicit(const iDynTree::FrameIndex refFrameIndex,
+                                                     const iDynTree::FrameIndex frameIndex,
+                                                     const iDynTree::FrameIndex expressedOriginFrameIndex,
+                                                     const iDynTree::FrameIndex expressedOrientationFrameIndex,
+                                                     iDynTree::MatrixDynSize & outJacobian)
+{
+    if (!pimpl->m_robot_model.isValidFrameIndex(frameIndex))
+    {
+        reportError("KinDynComputations","getRelativeJacobian","Frame index out of bounds");
+        return false;
+    }
+    if (!pimpl->m_robot_model.isValidFrameIndex(refFrameIndex))
+    {
+        reportError("KinDynComputations","getRelativeJacobian","Reference frame index out of bounds");
+        return false;
+    }
+    if (!pimpl->m_robot_model.isValidFrameIndex(expressedOriginFrameIndex))
+    {
+        reportError("KinDynComputations","getRelativeJacobian","expressedOrigin frame index out of bounds");
+        return false;
+    }
+    if (!pimpl->m_robot_model.isValidFrameIndex(expressedOrientationFrameIndex))
+    {
+        reportError("KinDynComputations","getRelativeJacobian","expressedOrientation frame index out of bounds");
+        return false;
+    }
+
+    //See Traversaro's PhD thesis, 3.37
+    // (with: D:= frame, L := refFrame)
+    //Given two links, D and L, the left-trivialized relative jacobian
+    //{}^D S_{L, D} (i.e. the relative jacobian of D w.r.t. L written in D.
+    // which yields the left-trivialized relative velocity {}^D V_{L,D},
+    // i.e. the relative velocity of D wrt L written in D and where
+    // {}^D V_{L,D} = {}^D S_{L, D} \dot{s}
+    //
+    // The jacobian can be computed column-wise as
+    // {}^D S_{L, D}_i =
+    // Case: - {}^D X_F {}^F s_{E,F}  if i \in \pi^{DOF}_L (D) and DoFOffset({E, F}) = i
+    //       - 0 otherwise
+    // where,
+    // {}^D X_F is the velocity transformation from F to D
+    // {}^F s_{E,F} (as velocity vector) also called joint motion subspace vector: Describes the
+    //              velocity of the joint motion, relative velocity of F w.r.t. E written in F.
+    // \pi^{DOF}_L (D) degrees of freedom in the path connecting the Link D to the link L (as if it were the base).
+
+
+    // compute fwd kinematics (if necessary)
+    this->computeFwdKinematics();
+
+    // Get the links to which the frames are attached
+    LinkIndex jacobianLinkIndex = pimpl->m_robot_model.getFrameLink(frameIndex);
+    LinkIndex refJacobianLink = pimpl->m_robot_model.getFrameLink(refFrameIndex);
+
+    //I have the two links. Create the jacobian
+    outJacobian.resize(6, pimpl->m_robot_model.getNrOfDOFs());
+    outJacobian.zero();
+
+    //move this to the traversal pool
```

Comment:
Opened an issue for this otherwise we will probably forget https://github.com/robotology/idyntree/issues/271 .